### ISSUE TITLE
Using System.Text.Json for assets file parsing is failing .NET SDK tests

### ISSUE THREAD

**Issue by nkolev92** [NuGet/Home#13248]
See: https://github.com/dotnet/sdk/pull/38774.

**Comment by zivkan** [comment]
I only checked one test, Microsoft.NET.Build.Tests.GivenThatWeWantDiagnosticsWhenAssetsFileCannotBeRead.It_reports_corrupt_file, but its failure is:

```text
 System.NullReferenceException: Object reference not set to an instance of an object.
    at NuGet.ProjectModel.LockFile.GetTarget(String frameworkAlias, String runtimeIdentifier)
    at Microsoft.NET.Build.Tasks.LockFileExtensions.GetTargetAndReturnNullIfNotFound(LockFile lockFile, String frameworkAlias, String runtimeIdentifier) in D:\src\sdk\src\Tasks\Microsoft.NET.Build.Tasks\LockFileExtensions.cs:line 14
```

My understanding is that this means `LockFileFormat` didn't not error out while loading the assets file, and now the SDK the is trying to pull information out of the `LockFile` data structure (via the `LockFile.GetTarget` method we provide), and it's throwing a null reference exception.

The test name makes it clear it's testing error scenarios, so it's not a "serious" failure (maybe the new code correctly handles all happy cases?), but the error handling appears to have changed with our STJ implementation.

**Comment by akoeplinger** [comment]
https://github.com/NuGet/NuGet.Client/pull/5637 fixed the `It_reports_corrupt_file` test, now we just have the other one remaining:

```
Microsoft.NET.Build.Tasks.UnitTests.GivenAResolvePackageDependenciesTask.ItAssignsDiagnosticLevel [FAIL]
Expected value to be 3, but found 0 (difference of -3).
  Stack Trace:
     at FluentAssertions.Execution.XUnit2TestFramework.Throw(String message)
     at FluentAssertions.Execution.TestFrameworkProvider.Throw(String message)
     at FluentAssertions.Execution.DefaultAssertionStrategy.HandleFailure(String message)
     at FluentAssertions.Execution.AssertionScope.FailWith(Func`1 failReasonFunc)
     at FluentAssertions.Execution.AssertionScope.FailWith(Func`1 failReasonFunc)
     at FluentAssertions.Execution.AssertionScope.FailWith(String message, Object[] args)
     at FluentAssertions.Numeric.NumericAssertions`2.Be(T expected, String because, Object[] becauseArgs)
  /_/src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenAResolvePackageDependenciesTask.cs(174,0): at Microsoft.NET.Build.Tasks.UnitTests.GivenAResolvePackageDependenciesTask.ItAssignsDiagnosticLevel()
     at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
     at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
```

**Comment by akoeplinger** [comment]
@zivkan @jgonz120 I extracted the test into a standalone repro:

```
var lockFile = new NuGet.ProjectModel.LockFileFormat().Read("failing.lock.json");
```

with this file:  [failing.lock.json](https://github.com/NuGet/Home/files/14363650/failing.lock.json)


It looks like it's failing to parse the file, e.g. Version is `-2147483648` instead of the expected `3`

Before:
<img width="654" alt="image" src="https://github.com/NuGet/Home/assets/1376924/4fab9d29-3b31-4e93-9e17-8a22e834970e">

After:
<img width="674" alt="image" src="https://github.com/NuGet/Home/assets/1376924/46024f24-e8f2-4c6c-82a0-040fb00c038a">

**Comment by jgonz120** [comment]
> @zivkan @jgonz120 I extracted the test into a standalone repro:
> 
> ```
> var lockFile = new NuGet.ProjectModel.LockFileFormat().Read("failing.lock.json");
> ```
> 
> with this file: [failing.lock.json](https://github.com/NuGet/Home/files/14363650/failing.lock.json)
> 
> It looks like it's failing to parse the file, e.g. Version is `-2147483648` instead of the expected `3`
> 
> Before: <img alt="image" width="654" src="https://private-user-images.githubusercontent.com/1376924/306734810-4fab9d29-3b31-4e93-9e17-8a22e834970e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg1NDA4MzEsIm5iZiI6MTcwODU0MDUzMSwicGF0aCI6Ii8xMzc2OTI0LzMwNjczNDgxMC00ZmFiOWQyOS0zYjMxLTRlOTMtOWUxNy04YTIyZTgzNDk3MGUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMjFUMTgzNTMxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjA1N2M5MDJmOTFjNjg2Nzg2MDc1MzJmODhlNWY4NzMwYTU2OWMwZWQzZmIwMmMyNDdjODg3NTMxYTNiMWE2ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.HXSIgpIRQpXGavjcvWya2SrO8mlfgxVB-FLF4FwUDF0">
> 
> After: <img alt="image" width="674" src="https://private-user-images.githubusercontent.com/1376924/306734910-46024f24-e8f2-4c6c-82a0-040fb00c038a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg1NDA4MzEsIm5iZiI6MTcwODU0MDUzMSwicGF0aCI6Ii8xMzc2OTI0LzMwNjczNDkxMC00NjAyNGYyNC1lOGYyLTRjNmMtODJhMC0wNDBmYjAwYzAzOGEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMjFUMTgzNTMxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGUzMjZmYTE2OGE5NWU5NWU5NjdiN2U1NjBiZjExOGFiNWUyYmEzODg0MzY2MmEwOTg5MDkwYzM1MDkwNWUzYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.dOll0OHej6liagoyzcN4XNE_9Ws6m7kN2zLsBi7RLAk">

The lock file contains a string for the warning level when it's supposed to be a number. The test needs to be updated to create a properly formatted file.

**Comment by zivkan** [comment]
@jgonz120 shouldn't `LockFileFormat` be throwing an format exception, rather than successfully returning an object with an invalid value?   I don't think it's just a test setup problem.

**Comment by akoeplinger** [comment]
I can confirm it parses when I change the warningLevel to be a number instead of a string so this should unblock the sdk code flow.

I agree though that throwing some exception during parsing would be better.

**Comment by jgonz120** [comment]
Agreed, but the previous code flow behaves this way so I feel like updating it would be out of scope for this change.

https://github.com/NuGet/NuGet.Client/blob/d49b8abb6b64d2a49a439380af8c62d099ba31fb/src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFileFormat.cs#L118

**Comment by akoeplinger** [comment]
Interesting. That also means code like this doesn't make sense since we'll never get an exception: https://github.com/dotnet/sdk/blob/302ca4c93537954c056289cde3a2c4faac0d9ce2/src/Cli/Microsoft.DotNet.Cli.Utils/Extensions/LockFileFormatExtensions.cs#L25

**Comment by jgonz120** [comment]
Yea seems like it. In the LockFileCache class there's a logger that throws an error when LogError is called, something like that could be used to make that code make sense.  https://github.com/dotnet/sdk/blob/302ca4c93537954c056289cde3a2c4faac0d9ce2/src/Tasks/Microsoft.NET.Build.Tasks/LockFileCache.cs#L79

**Comment by akoeplinger** [comment]
Actually I take that back, this calls `public LockFile Read(string filePath)` which uses `File.OpenRead(filePath)` before passing the stream to the other Read method so we'd catch the exception from there.

I think we can close this issue.
